cl/lib/pacer.py

```diff
@@ -530,3 +534,118 @@ def normalize_attorney_contact(c, fallback_name=""):
     address_info = normalize_address_info(dict(address_info))
     address_info["lookup_key"] = make_address_lookup_key(address_info)
     return address_info, atty_info
+
+
+class ConnectionType(TypedDict):
+    connection_ok: bool
+    status_code: int | None
+    date_time: datetime
+
+
+def check_pacer_court_connectivity(court_id: str) -> ConnectionType:
+    """Check if PACER is accessible for the given court.
+
+    :param court_id: The court ID to check.
+    :returns: A dict with the court connection status.
+    """
+
+    url = f"https://ecf.{court_id}.uscourts.gov/"
+
+    connection_ok = False
+    status_code = None
+    try:
+        r = requests.get(url, timeout=5)
+        if r.status_code == requests.codes.ok:
```

Comment:
Would it be better to use `raise_for_status()` here?

Reply:
yes, I've added raise_for_status(), so only status codes below 400 are going to set `connection_to  = True`